pdfquery/pdfquery.py

```diff
@@ -546,6 +570,35 @@ def _cached_pages(self, target_page=-1):
         self._pages += list(self._pages_iter)
         return self._pages
 
+    def _add_annots(self, layout, annots):
+        """Adds annotations to the layout object
+        """
+        if annots:
+            for annot in annots:
+                annot = annot.resolve()
+                annot['bbox'] = annot.pop('Rect')  # Rename key
+                annot = self._set_hwxy_attrs(annot)
+                annot['URI'] = annot['A']['URI']

```

Comment:
Here we should test for a 'Rect' key before lines 579-580, and for both an 'A' and 'URI' key before line 581.

(Partly because some kinds of annotations intentionally lack those things, and partly because there are lots of malformed PDFs out there and assuming that PDFs will have any given key is just inviting trouble. :)